### ISSUE TITLE
First search, then freeze 

### ISSUE THREAD

**Issue by TomGriebenow** [owncloud/music#945]
In version 1.5.1 the computer freezes when new music is searched for via the scan. Is the error known?

**Comment by paulijar** [comment]
Nope, there's no known issue like that. You need to provide some more details, like which version of ownCloud or Nextcloud are you using, and are you talking about the server computer or the client computer. If it is the client computer, then which browser are you using? And what exactly you mean by freezing in this context? For me, it would mean that the whole computer would become unresponsive and you would need to power it off to recover, but you might mean something different.

**Comment by TomGriebenow** [comment]
OK, sorry, I hadn't thought of that!
I'm running Nextcloud 22.2.3 on a Linux server (Kernel 5.4.0-88 generic x86_64), PHP 8.0.12 and MySQL 10.5.12.
I use the client to access the music module via the website (Firefox on Linux), it shows that unlisted music is still available and when updating, the server freezes when the last piece of music has been recognized. Then I have to restart the server.

**Comment by paulijar** [comment]
Okay, thanks for the info. If the freezing happens after running through all the audio files, then it might happen when the app tries to find image files to use as cover art for the albums and artists. 

Does your Nextcloud maybe contain exceptionally large amount of image files? At least a couple of years ago Nextcloud and ownCloud were rather brain-dead in how they handled the searching of files by MIME type under a specified folder, and it became slower and slower when the amount of files of the searched type increased anywhere within the user account. That is, searching e.g. image files from even an empty folder could take minutes or hours if there were hundreds of thousands of images in some other folders. I'm not certain if the situation has improved since, but if it hasn't, then that could jam the server pretty badly.

Also, can you find anything from nextcloud.log after the problem has happened?

**Comment by paulijar** [comment]
Oh, and one more thing. Have you tried scanning from the command line using `occ music:scan`? I assume that you probably encounter the same freeze-up there, but it could confirm or disprove the theory that the hang-up happens while searching the images: the command-line scanning prints out the paths of the files being scanned and when it starts to search for the cover images.

**Comment by TomGriebenow** [comment]
I tried various settings to see how the app behaves. When I excluded the attached external storage (same provider, different server), the error no longer occurred! Probably the access between the computers took too long.

**Comment by wKollendorf** [comment]
> I tried various settings to see how the app behaves. When I excluded the attached external storage (same provider, different server), the error no longer occurred! Probably the access between the computers took too long.

Hi, I encountered the same behaviour. When searching external storage, the server needs to be restarted. Also the used postgresql DB is flooded with open connections without closing them.
